### ISSUE TITLE
Have the ability to name each step in the pipeline and use that in complex pipelines

### ISSUE THREAD

**Issue by husamrahman** [Mu-Sigma/analysis-pipelines#51]
Feature Request: Have the ability to name the steps in the pipeline and reference them in complex pipelines.

For example:

Current State: function( x = ~f1) where f1 comes from some previous function
Feature Request: function( x = ~someName) where someName would be the name of the step in the pipeline defined by the user

**Comment by naren1991** [comment]
Yes, makes sense. It was something I had considered when initially designing, but chose the 'f1' syntax for more concise writing. However, allowing the user to use the name of the function in the formula does make it more readable.

Will reconsider this choice, evaluate the impact of adding this feature and consider adding in a future release.

**Comment by husamrahman** [comment]
Great! You could keep the default naming convention as "f`1" to have backwards compatibility if you do decide to make the change. Fantastic package none the less!

**Comment by naren1991** [comment]
Thank you! And thank you for all your feedback!

 Keeping this issue open until then.
